### PR TITLE
Further refactoring in preparation for CONST_INTEGER

### PR DESCRIPTION
**Pull request by ajreynol**
Miscellaneous refactorings from trying to enable CONST_INTEGER.

### CODE REVIEW

File: src/theory/arith/linear/normal_form.cpp [hunk 1]
```diff
@@ -249,13 +252,16 @@ Monomial Monomial::mkMonomial(const VarList& vl) {
 }
 
 Monomial Monomial::parseMonomial(Node n) {
-  if(n.getKind() == kind::CONST_RATIONAL) {
+  Kind k = n.getKind();
+  if (k == kind::CONST_RATIONAL || k == kind::CONST_INTEGER)
```
Comment (nafur) [inline src/theory/arith/linear/normal_form.cpp:256]: Should we use `isConst()` instead here?

File: src/theory/arith/linear/normal_form.cpp [hunk 2]
```diff
@@ -914,19 +923,20 @@ Node Comparison::toNode(Kind k, const Polynomial& l, const Polynomial& r) {
...
 bool Comparison::rightIsConstant() const {
+  Kind k;
   if(getNode().getKind() == kind::NOT){
-    return getNode()[0][1].getKind() == kind::CONST_RATIONAL;
+    k = getNode()[0][1].getKind();
   }else{
-    return getNode()[1].getKind() == kind::CONST_RATIONAL;
+    k = getNode()[1].getKind();
   }
+  return k == kind::CONST_RATIONAL || k == kind::CONST_INTEGER;
```
Comment (nafur) [inline src/theory/arith/linear/normal_form.cpp:939]: also use `isConst()` here?

**Review by nafur — APPROVED** [review]
two comments, otherwise LGTM

**Comment by ajreynol** [comment]
I think I'd prefer checking the kinds, as this also ensures the types are correct. I'm not sure if the code is already safe for this.  Also the check on kinds may be slightly faster.

**Pull request merged**